lib/seadsa/RemovePtrToInt.cc

```diff
@@ -440,7 +440,12 @@ bool RemovePtrToInt::runOnFunction(Function &F) {
       llvm::make_filter_range(MaybeUnusedInsts, [](Instruction *I) {
         return isInstructionTriviallyDead(I);
       }));
-  RecursivelyDeleteTriviallyDeadInstructions(TriviallyDeadInstructions);
+  
+  SmallVector<WeakTrackingVH, 16> TriviallyDeadHandles;
```

Comment:
comment why this is needed. I don't think we have code like this before. Is this something new about LLVM11?

Reply:
Yes, LLVM 11 changes the signature of function `RecursivelyDeleteTriviallyDeadInstructions`. One of its new signatures is as follows,

```C++
void llvm::RecursivelyDeleteTriviallyDeadInstructions (SmallVectorImpl< WeakTrackingVH > &DeadInsts, const TargetLibraryInfo *TLI=nullptr, MemorySSAUpdater *MSSAU=nullptr, std::function< void(Value *)> AboutToDeleteCallback=std::function< void(Value *)>())
```

Reply:
maybeu used instructions should be weak handles. Basically the new rules is to keep handles rather than instructions as we did before. This is really nice because it allows the instructions to be deleted any time.

Reply:
> At least comment that this is a big TODO item. I do not have LLVM11 installed locally, so cannot change and check myself. `MaybeUnusedInsts` has to a collection of `WeakTrackingVH`, not instructions.

Sorry, I'm confused by your last comment. I think the existing implementation looks fine because none of the instructions in `MaybeUnusedInsts` are removed before calling `RecursivelyDeleteTriviallyDeadInstructions` on them.

Reply:
It would be better in existing implementation to use ValueHandle instead of raw instructions. It would have solved lots of issues we ran into. I simply did not know of these :)

If this is blocking you, just please add a comment around the code explaining why it is this way and we can move on.

Reply:
> It would be better in existing implementation to use ValueHandle instead of raw instructions. It would have solved lots of issues we ran into. I simply did not know of these :)
> 
> If this is blocking you, just please add a comment around the code explaining why it is this way and we can move on.

Thank you for your patience. I added a comment so we're not blocked by this issue. I'd be happy to revisit this issue once LLVM 11 support is done.